mobile/library/common/apple/utility.cc

```diff
@@ -0,0 +1,38 @@
+#include "library/common/apple/utility.h"
+
+namespace Envoy {
+namespace Apple {
+
+std::string toString(CFStringRef string) {
+  // A pointer to a C string or NULL if the internal storage of string
+  // does not allow this to be returned efficiently.
+  auto efficient_c_str = CFStringGetCStringPtr(string, kCFStringEncodingUTF8);
+  if (efficient_c_str) {
+    return std::string(efficient_c_str);
+  }
+
+  CFIndex length = CFStringGetLength(string);
+  CFIndex size = CFStringGetMaximumSizeForEncoding(length, kCFStringEncodingUTF8) + 1;
+  char* c_str = static_cast<char*>(malloc(size));
```

Comment:
can we unique_ptr this? I think we're missing a free if we fail the getCString call